internal/pkg/configmanager/configmanager.go

```diff
@@ -18,31 +18,86 @@ var (
 	GOARCH string = runtime.GOARCH
 )
 
-// Config is the struct for loading DevStream configuration YAML files.
-// It records rendered config values and is used as a general config in DevStream.
+// Config records rendered config values and is used as a general config in DevStream.
 type Config struct {
 	Tools []Tool `yaml:"tools"`
 	State *State
 }
 
+func (c *Config) Validate() []error {
+	errors := make([]error, 0)
+
+	for _, t := range c.Tools {
+		errors = append(errors, t.Validate()...)
+	}
+
+	errors = append(errors, c.ValidateDependency()...)
+
+	return errors
+}
+
+func (c *Config) ValidateDependency() []error {
+	errors := make([]error, 0)
+
+	// config "set" (map)
+	toolMap := make(map[string]bool)
+	// creating the set
+	for _, tool := range c.Tools {
+		toolMap[tool.Key()] = true
+	}
+
+	for _, tool := range c.Tools {
+		// no dependency, pass
+		if len(tool.DependsOn) == 0 {
+			continue
+		}
+
+		// for each dependency
+		for _, dependency := range tool.DependsOn {
+			// skip empty string
+			dependency = strings.TrimSpace(dependency)
+			if dependency == "" {
+				continue
+			}
+
+			// generate an error if the dependency isn't in the config set,
+			if _, ok := toolMap[dependency]; !ok {
+				errors = append(errors, fmt.Errorf("tool %s's dependency %s doesn't exist in the config", tool.InstanceID, dependency))
+			}
+		}
+	}
+
+	return errors
+}
+
+type Manager struct {
+	ConfigFile string
+}
+
+func NewManager(ConfigFileName string) *Manager {
```

Comment:
Maybe unexported var `configFileName` is more suitable.

Reply:
You could also fix it in next pr.

Reply:
right! I'll fix it in no time.